test_util/cluster_api.py

```diff
@@ -87,6 +87,24 @@ def _wait_for_dcos_history_up(self):
     @retrying.retry(wait_fixed=1000,
                     retry_on_result=lambda ret: ret is False,
                     retry_on_exception=lambda x: False)
+    def _wait_for_dcos_history_data(self):
+        for _ in range(10):
+            # call 10 times to make sure you get something, might need to sleep instead
+            ro = self.get('/dcos-history-service/history/last')
+        # resp_code >= 500 -> backend is still down probably
+        if ro.status_code <= 500:
+            logging.info("DC/OS History is probably getting data")
+            json = ro.json()
+            assert len(json["slaves"]) != 0
```

Comment:
I would prefer that we just assert that we get all the expected slaves rather than only asserting nonzero, but I could also see the argument that this waiting-style check should not make specific assertions.